cockatrice/src/dialogs/dlg_move_top_cards_until.h

```diff
@@ -0,0 +1,26 @@
+#ifndef DLG_MOVE_TOP_CARDS_UNTIL_H
+#define DLG_MOVE_TOP_CARDS_UNTIL_H
+
+#include <QDialog>
+#include <QDialogButtonBox>
+#include <QLabel>
+#include <QLineEdit>
+#include <QSpinBox>
+#include <QString>
+
+class DlgMoveTopCardsUntil : public QDialog
+{
+    Q_OBJECT
+
+    QLabel *exprLabel, *numberOfHitsLabel;
+    QLineEdit *exprEdit;
+    QSpinBox *numberOfHitsEdit;
+    QDialogButtonBox *buttonBox;
+
+public:
+    explicit DlgMoveTopCardsUntil(QWidget *parent = nullptr, QString expr = QString(), uint numberOfHits = 1);
+    [[nodiscard]] QString getExpr() const;
```

Comment:
I don't think there is a lot of benefit of adding `nodiscard` to a simple getter, while yeah you probably wouldn't want to discard the return value from this function, you also wouldn't call it if you didn't use the return value, if we were to use `nodiscard` like this we might as well add it to every return value which would just make the code harder to read, imo `nodiscard` is to be used only for functions that are called for their side effects that would return for example an important error value.

Reply:
i just copied the code from `DlgRollDice` lol. But sure

Reply:
hmm, I wonder who wrote that now

Reply:
The nodiscard is a big recommendation from clion nowadays. Not sure why, but i've been adding it in as appropriate

Reply:
I disagree with it, clion is adding things we don't need and it seems like clion is just pretending to be smart :(